integration/spark/shared/src/main/java/io/openlineage/spark/agent/util/JdbcUtils.java

```diff
@@ -43,8 +43,10 @@ public static Optional<SqlMeta> extractQueryFromSpark(JDBCRelation relation) {
               Collections.emptyList()));
     } else {
       String query =
-          tableOrQuery.replaceFirst("\\(", "").replaceAll("\\) SPARK_GEN_SUBQ_[0-9]+", "");
-
+          tableOrQuery
+              .substring(0, tableOrQuery.lastIndexOf(")"))
+              .replaceFirst("\\(", "")
+              .replaceAll("`", "");
```

Comment:
We're using `sql-parser` library to parse queries and the library is capable of parsing backtick quoted identifiers, at least for databases that support it, like mysql: 
https://github.com/sqlparser-rs/sqlparser-rs/blob/main/src/dialect/mysql.rs#L35

Are we able to pass dialect information information from Spark into the parser? This could prevent other similar issues in future. If this is a bigger change, we should at least have an open issue for that.

Reply:
@pawel-big-lebowski yes, I've extracted the dialect from the url, when using the parse method with dialect it works fine with the backticks :)

Reply:
@nataliezeller1 please revisit the failing tests:
```
  java.lang.NullPointerException
      at io.openlineage.spark3.agent.lifecycle.plan.column.JdbcColumnLineageExpressionCollectorTest.testInputCollection(JdbcColumnLineageExpressionCollectorTest.java:68)
```

Reply:
fixed the failing tests